integration-cli/docker_api_swarm_service_test.go

```diff
@@ -596,3 +596,68 @@ func (s *DockerSwarmSuite) TestAPISwarmServicesStateReporting(c *check.C) {
 		}
 	}
 }
+
+// Test plugins deployed via swarm services
+func (s *DockerSwarmSuite) TestAPISwarmServicesPlugin(c *check.C) {
+	testRequires(c, DaemonIsLinux, IsAmd64)
+
+	d1 := s.AddDaemon(c, true, true)
+	d2 := s.AddDaemon(c, true, true)
+	d3 := s.AddDaemon(c, true, false)
+
+	time.Sleep(1 * time.Second) // make sure all daemons are ready to accept
+
+	makePlugin := func(s *swarm.Service) {
+		s.Spec.TaskTemplate.Runtime = "plugin"
+		s.Spec.TaskTemplate.PluginSpec = &swarm.PluginSpec{
+			Name:   "test",
+			Remote: "cpuguy83/docker-logdriver-test",
```

Comment:
I'm hoping we can avoid adding new dependencies on Docker Hub in tests; see https://github.com/moby/moby/issues/30626